model/task_lifecycle.go

```diff
@@ -1184,25 +1192,50 @@ func UpdateDisplayTask(t *task.Task) error {
 	return nil
 }
 
+func checkResetSingleHostTaskGroup(t *task.Task, caller string) error {
```

Comment:
are display tasks ever in single-host task groups?

Reply:
It looks to me like we set task group information on just the [execution tasks](https://github.com/evergreen-ci/evergreen/blob/de615ac07c5c256b0d3c339baa371aa582677ca4/model/lifecycle.go#L1139) and not [display tasks](https://github.com/evergreen-ci/evergreen/blob/de615ac07c5c256b0d3c339baa371aa582677ca4/model/lifecycle.go#L1144)

Reply:
Oh nice find, then yes things should be good!